### ISSUE TITLE
`max-statements-per-line`: FunctionDeclaration still throws error if line doesn't have let/const

### ISSUE THREAD

**Issue by Pecacheu** [eslint-stylistic/eslint-stylistic#1198]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Configuration:
```js
import js from "@eslint/js";
import { defineConfig } from "eslint/config";
import sty from "@stylistic/eslint-plugin";

export default defineConfig([
	{
		files: ["**/*.{js,mjs,cjs,ts,mts,cts}"],
		ignores: ["build.js", "dist/**"],
		plugins: {js, sty},
		rules: {
			"sty/max-statements-per-line": ["warn", {ignoredNodes: [
				'BreakStatement',
				'ClassDeclaration',
				'ContinueStatement',
				'DebuggerStatement',
				'DoWhileStatement',
				'ForInStatement',
				'ForOfStatement',
				'ForStatement',
				'FunctionDeclaration',
				'IfStatement',
				'ImportDeclaration',
				'LabeledStatement',
				'ReturnStatement',
				'SwitchStatement',
				'ThrowStatement',
				'TryStatement',
				'VariableDeclaration',
				'WhileStatement',
				'WithStatement',
				'ExportNamedDeclaration',
				'ExportDefaultDeclaration',
				'ExportAllDeclaration'
			]}]
		}
	}
]);
```

```js
let test = () => {console.log("Stuff")}; //No error
test = () => {console.log("Stuff 2.0")}; //Error
```

This creates a problem for example if trying to write something that could easily be a one-liner like:
```js
setTimeout(() => {variable = null}, 500);
```

Second thing, there is also no way to allow if/else on same line with this rule, eg:
```js
if(a) doThing(); else doOtherThing();
```

I assumed ignoredNodes `IfStatement` would do this, but it has no effect.

Really my only goal in using this rule was to prevent misleading situations like this, while leaving anything else untouched:
```js
if(a) doThing(); thisIsOutsideTheIf();
```

### Reproduction

Your Stackblitz template is still broken btw

### System Info

```shell
System:
    OS: Windows 11 10.0.26200
  Binaries:
    Node: 24.4.1 - C:\Program Files\nodejs\node.EXE
    npm: 11.4.2 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    @stylistic/eslint-plugin: ^5.10.0 => 5.10.0
    eslint: ^10.0.3 => 10.1.0
```

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests
